app/Http/Controllers/AuthenticationController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Str;
+use App\Enums\UserStatus;
+use App\Models\Authentication;
+use App\Mail\SendTokenMail;
+use App\Http\Requests\AuthenticationRequest;
+use Carbon\Carbon;
+
+class AuthenticationController extends Controller
+{
+    public function create(Request $request)
+    {
+        return view('authentication/create');
+    }
+    public function store(AuthenticationRequest $request)
+    {
+        $user_token = Str::random(rand(30, 50));
+        $authentication = Authentication::where('email', $request->email)
+            ->where('status',UserStatus::MAIL_SENT)
+            ->first();
+
+        if(!is_null($authentication)){
+            $authentication->token = $user_token;
+            $authentication->expiration_at = Carbon::now()->addMinutes(15);
+            $authentication->save();
+        }else{
+            Authentication::create([
+                'token' => $user_token,
+                'email' => $request->email,
+                'status' => UserStatus::MAIL_SENT,
+                'expiration_at' => Carbon::now()->addMinutes(15)
+            ]);
+        }
+
+        Mail::to($request->email)->send(new SendTokenMail(route('users.create', $user_token)));
+
+        return to_route('authentications.complete', $user_token)->with(['authentications_complete' => true]);
```

Comment:
```suggestion
        return to_route('authentications.complete', $user_token)->with(['is_authentication_created' => true]);
```
bool値を代入する物にはこのような命名をします。
Is authentication created ? のような英文がyes or noの二択で返答するのが由来です。

Reply:
ご指摘、ご説明、命名のご提案ありがとうございます。
is_authentication_createdへと修正致しました。
ご確認宜しくお願い致します。
app/Http/Controllers/AuthenticationController.php